### ISSUE TITLE
Renaming files on Windows machine can lead to deletion

### ISSUE THREAD

**Issue by bennyborn** [contao/core#8442]
By accident I just figured out that it's not possible to rename a file with a mixed-case naming to all lower- or all uppercase due to a "fix" in  [/system/modules/core/library/Contao/Files/Php.php#L121-L125](https://github.com/contao/core/blob/436f0cc99089756e7a3d2e0aa53155b1823942d6/system/modules/core/library/Contao/Files/Php.php#L121-L125).

This "fix" deletes the file and tries to rename it afterwards...

Maybe there should be a check like this to prevent the deletion if the filename is the same regardless of case-sensitivity:

```php
// Windows fix: delete the target file
if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN' && file_exists(TL_ROOT . '/' . $strNewName) && strcasecmp($strOldName, $strNewName) !== 0)
{
    $this->delete($strNewName);
}
```

**Comment by leofeyer** [comment]
Can anyone reproduce this?

**Comment by asaage** [comment]
i can reproduce this in 3.5.12
renamed:
```CamelCase.png``` to  ```camelcase2.png``` : everything fine
```CamelCase.png``` to  ```camelcase.png``` : error and the file gets deleted

log:
```
Details:	File or folder "files/camelcase.png" was not mounted or could not be found
Funktion:	Contao\DC_Folder::edit
 ```

but no hint in ```system/logs/error.log```

**Comment by leofeyer** [comment]
Fixed in af4e01326dc8d915cfede9c333a1c0b8a5e641ff.
